crates/rune/src/languageserver/state.rs

```diff
@@ -191,6 +193,205 @@ impl<'a> State<'a> {
         Some(location)
     }
 
+    /// Find definition at the given uri and LSP position.
+    pub async fn complete(
+        &self,
+        uri: &Url,
+        position: lsp::Position,
+    ) -> Option<Vec<lsp::CompletionItem>> {
+        let sources = &self.workspace.sources;
+        tracing::info!("sources: {:?}", sources.get(uri).is_some());
+        let workspace_source = sources.get(uri)?;
+
+        let offset = workspace_source.lsp_position_to_offset(position);
+
+        let (mut symbol, start) = workspace_source.looking_back(offset)?;
+        tracing::info!("symbol : {:?}, start: {:}", symbol, start);
+
+        if symbol.is_empty() {
+            return None;
+        }
+
+        let mut results = vec![];
+
+        let can_use_instance_fn: &[_] = &['.'];
```

Comment:
Should probably just check `== '.'` below. Though this is a bit clearer documentation-wise.